File: anavi-gardening-uhat/led/python/blink.py
# coding=utf-8
import signal
import sys
import time

import RPi.GPIO as GPIO

# Pin 15 on Raspberry Pi corresponds to GPIO 22
LED1 = 15
# Pin 16 on Raspberry Pi corresponds to GPIO 23
LED2 = 16


def close(signal, frame):
    GPIO.output(LED1, 0)
    GPIO.output(LED2, 0)
    sys.exit(0)


signal.signal(signal.SIGINT, close)

# to use Raspberry Pi board pin numbers
GPIO.setmode(GPIO.BOARD)
GPIO.setwarnings(False)

# set up GPIO output channel
GPIO.setup(LED1, GPIO.OUT)
GPIO.setup(LED2, GPIO.OUT)

# reset

try:
    while True:
        GPIO.output(LED1, 1)
        GPIO.output(LED2, 0)
        time.sleep(0.5)
        GPIO.output(LED1, 0)
        GPIO.output(LED2, 1)
        time.sleep(0.5)
finally:
    GPIO.cleanup()

File: anavi-gardening-uhat/led/c/blink.c
#include <stdio.h>
#include <signal.h>
#include <stdlib.h>
#include <pigpio.h>

//Pin 15 on Raspberry Pi corresponds to BCM GPIO 22
#define D1 22
//Pin 16 on Raspberry Pi corresponds to BCM GPIO 23
#define D2 23

void terminate()
{
	gpioWrite(D1, 0);
	gpioWrite(D2, 0);

	//Stop DMA, release resources
	gpioTerminate();
	exit(0);
}

int main()
{
	if (0 > gpioInitialise())
	{
		fprintf(stderr, "pigpio initialisation failed\n");
		return 1;
	}

	// Set GPIOs to output modes
	gpioSetMode(D1, PI_OUTPUT);
	gpioSetMode(D2, PI_OUTPUT);

	signal(SIGINT, terminate);

	// Blink
	while(1)
	{
		gpioWrite(D1, 1);
		gpioWrite(D2, 0);
		time_sleep(0.5);
		gpioWrite(D1, 0);
		gpioWrite(D2, 1);
		time_sleep(0.5);
	}

	// Stop DMA, release resources
	gpioTerminate();

	return 0;
}
